cpp/src/io/utilities/datasource.cpp

```diff
@@ -98,6 +98,8 @@ class file_source : public datasource {
  private:
   std::unique_ptr<detail::cufile_input_impl> _cufile_in;
   kvikio::FileHandle _kvikio_file;
+  // The read size above which GDS is faster then posix-read + h2d-copy
+  static constexpr size_t _gds_io_preferred_threshold = 128 << 10;
```

Comment:
This is bad. You have the same constant defined in two separate places without sync.

Please define it in one place and use it in both.

Reply:
These don't really represent the same value. We could potentially use different thresholds for reads and writes.

Reply:
Then please use different names. Using exactly the same name will cause confusion.

Reply:
Also:
```suggestion
  static constexpr size_t _gds_io_preferred_threshold = 128 << 10; // 128KB
```